Source/WebCore/dom/Element.cpp

```diff
@@ -3097,13 +2894,9 @@ Node::InsertedIntoAncestorResult Element::insertedIntoAncestor(InsertionType ins
     } else if (!hasLanguageAttribute())
         updateEffectiveLangStateFromParent();
 
-    if (RefPtr parent = parentOrShadowHostElement(); parent && UNLIKELY(parent->usesEffectiveTextDirection())) {
-        auto* input = dynamicDowncast<HTMLInputElement>(*this);
-        if (!elementAffectsDirectionality(*this) && !(input && input->isTelephoneField())) {
-            setUsesEffectiveTextDirection(true);
-            setEffectiveTextDirection(parent->effectiveTextDirection());
-        }
-    }
+    RefPtr parent = parentOrShadowHostElement();
+    if (UNLIKELY(selfOrPrecedingNodesAffectDirAuto() || (parent && parent->usesTextDirection())))
```

Comment:
I think this code was clearer before with the name "effective text direction".

Reply:
I was trying not to use different names. The difference between "text direction" and "effective text direction" is very subtle. I really do not understand what "effective" mean in this context. For me "effective" can be used when something is accumulated on top of each other. For example if you store "transparency" for every node in a tree. And the transparency of a node affects the transparency of its children. EffectiveTransparency, for me, will mean the accumulated transparency of a certain node from the root till we reach this node.

I do not see this analogy with the text direction. But I will revert this change.

Reply:
Done.